plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessor.java

```diff
@@ -810,7 +811,7 @@ public Answer createVolume(final CreateObjectCommand cmd) {
             final SR poolSr = hypervisorResource.getStorageRepository(conn,
                     CitrixHelper.getSRNameLabel(primaryStore.getUuid(), primaryStore.getPoolType(), primaryStore.getPath()));
             VDI.Record vdir = new VDI.Record();
-            vdir.nameLabel = volume.getName();
+            vdir.nameLabel = getEncodedVolumeName(volume.getName());
```

Comment:
@Pearl1594 can you check if this change is needed here as well (while creating volume from snapshot) =>

https://github.com/apache/cloudstack/blob/603cd84b49428c08fae50470c8c69a66468b53fe/plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/XenServerStorageProcessor.java#L510-L517

Reply:
In both places, the nameLabel is set to "cloud-<randomUUID>" before calling createVdi - so it shouldn't be an issue. I created a volume from a snapshot using a volume name that has an emoji and it created the following vdi:
![image](https://github.com/user-attachments/assets/b26de6ab-99f0-44dd-828e-5ee13f445ccd)
![image](https://github.com/user-attachments/assets/0ebe5fd5-90ea-44f3-8835-abbb062c353f)

Reply:
ok @Pearl1594 thanks for checking it.